assets/js/atomic/blocks/product-elements/price/block.tsx

```diff
@@ -101,4 +113,9 @@ export const Block = ( props: Props ): JSX.Element | null => {
 	);
 };
 
-export default withProductDataContext( Block );
+export default ( props: Props ) => {
+	if ( props.isDescendentOfSingleProductTemplate ) {
+		return <Block { ...props } />;
+	}
+	return withProductDataContext( Block )( props );
+};
```

Comment:
It's just a minor suggestion. In PR description, you have explained that `atomic blocks support several contexts`. Maybe we can add similar comment here to explain why we are returning early here?
Because it was not obvious to me until I looked at the PR description. What do you think? 🙂

Reply:
Good point! Fixed via 9d8a43a60dcb16bda85c5dde508bebc63c9b5f79